platform/platform.go

```diff
@@ -299,14 +299,35 @@ func CheckMachine(ctx context.Context, m Machine) error {
 		return fmt.Errorf("ssh unreachable: %v", err)
 	}
 
-	// ensure we're talking to a Container Linux system
 	out, stderr, err := m.SSH("grep ^ID= /etc/os-release")
 	if err != nil {
 		return fmt.Errorf("no /etc/os-release file: %v: %s", err, stderr)
 	}
 
-	if !bytes.Equal(out, []byte("ID=coreos")) && !bytes.Equal(out, []byte(`ID="rhcos"`)) {
-		return fmt.Errorf("not a supported instance")
+	supportedSystemCheck := func() (bool, error) {
+		// ensure we're talking to a supported system
+		switch string(out) {
+		case `ID=coreos`, `ID="rhcos"`:
+			return true, nil
+		case `ID=fedora`:
+			variantid, _, err := m.SSH("grep ^VARIANT_ID= /etc/os-release")
```

Comment:
I'd do `. /etc/os-release && echo "$VARIANT_ID"` - the file is shell syntax, and your "parser" here doesn't handle quotes.

Reply:
Also worth noting this obsoletes https://github.com/coreos/mantle/pull/920